keps/sig-node/4742-node-topology-downward-api/README.md

```diff
@@ -0,0 +1,411 @@
+
+#### Beta
+
+TODO after Alpha.
+
+#### GA
+
+TODO after Beta.
+
+### Upgrade / Downgrade Strategy
+
+Upgrade should be relatively safe because there no behavior changes unless topology labels
+are specified in the downward API.
+
+Already running Pods will not be affected by enabling this feature in kube-apiserver.
+
+There are some risks to downgrade, for example, if a Pod relies on the topology
+information using downward API, that information could get lost on downgrade.
+
+### Version Skew Strategy
+
+There should be no risks with version skew between control plane and nodes since there's no changes made to the kubelet's downward API implementation.
+All changes proposed in this KEP are control-plane only.
+
+For n-3 kubelets, old kubelets still support consuming Pod labels via downward API. This capability should be supported as long as the control plane
+has this feature enabled.
+
+There may be some risks for version skew with other components (e.g CSI or CNI) if the downward API is used
+to pass topology information to those components. Components should ensure they are using a kube-apiserver version
+that supports copying topology labels to Pods.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+A feature gate `PodTopologyLabelsAdmission` will be added to the kube-apiserver to enable or disable this feature.
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [X] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: `PodTopologyLabelsAdmission`
+  - Components depending on the feature gate: `kube-apiserver`
+- [ ] Other
+  - Describe the mechanism:
+  - Will enabling / disabling the feature require downtime of the control
+    plane?
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node?
+
+###### Does enabling the feature change any default behavior?
+
+Yes, all Pods will have topology labels copied over.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes it can, but Pods created while the feature is enabled will continue to have topology labels.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+New Pods will have topology labels copied over. Older Pods created while the feature was disabled will be missing topology labels.
+
+###### Are there any tests for feature enablement/disablement?
+
+Tests will be added to ensure feature gate works as expected.
+
+### Rollout, Upgrade and Rollback Planning
+
+TODO for Beta.
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+Failure on rollout is unlikely since we do not override any existing topolgoy labels on Pods.
+Already running workloads that is already copying topology labels to Pods should be untouched by the new admission plugin.
+
+###### What specific metrics should inform a rollback?
+
+N/A
```

Comment:
Could a bug in your code cause binding failures? Is there a metric to track binding failures?

Reply:
added one of the scheduler metrics here